projects/swimlane/ngx-cron/src/lib/ngx-cron.service.ts

```diff
@@ -215,35 +264,49 @@ export class NgxCronService {
     this.timezone = timezone;
   }
 
-  private validate(data: any): boolean {
-    if (data.min || data.minuteInterval > 59 || data.minuteInterval === 0) {
-      return false;
-    }
-    if (data.hour > 24) {
-      return false;
-    }
+  validateCronExpression(cron, isQuartz, configOverrides) {
+    const config = this.cronValidateConfig;
 
-    if (data.secondInterval > 59 || data.secondInterval === 0) {
-      return false;
+    config.override = { useSeconds: isQuartz, ...NgxCronService.CRON_VALIDATE_CONFIG_OVERRIDES, ...configOverrides };
+
+    // coerce cron to string when undefined or null to prevent CronValidate from failing
+    if (cron == null) {
+      cron = '';
     }
 
-    if (data.minute > 59) {
-      return false;
+    const cronResult = CronValidate(cron, config);
+
+    if (cronResult.isError()) {
+      const [error] = cronResult.getError();
+      return { isValid: false, error };
     }
+  }
+
+  private validate(data: any, cron, configOverrides) {
+    const err = this.validateCronExpression(cron, data.isQuartz, configOverrides);
+    if (err) return err;
 
-    if (data.weekday === undefined) {
-      return false;
+    if (data.minuteInterval > 59 || data.minuteInterval === 0 || data.minute > 59) {
+      const error = this.errorDescription('minutes', cron);
+      return { isValid: false, error };
     }
 
-    if (data.month === undefined) {
-      return false;
+    if (data.secondInterval > 59 || data.secondInterval === 0) {
+      const error = this.errorDescription('seconds', cron);
+      return { isValid: false, error };
     }
 
     const daysMax = data.daysMax;
     if ((daysMax !== null && data.day > daysMax) || data.day > 31 || data.day === 0) {
-      return false;
+      const error = `Day of the month does not exist for current month selected (Input cron: ${cron})`;
+      return { isValid: false, error };
     }
-    return true;
+
+    return { isValid: true, error: null };
+  }
+
+  private errorDescription(period, cron) {
```

Comment:
suggestion: `errorDescription` with `1 and 59` seems to be applied to minute and seconds only? Can we then rename `errorDescription` to something more specific? Can we also type `period` with a type-alias `"seconds" | "minutes"` to prevent misuse?